Branch-SDK/Branch-SDK/Requests/BranchOpenRequest.m

```diff
@@ -82,6 +82,7 @@ - (void)makeRequest:(BNCServerInterface *)serverInterface key:(NSString *)key ca
 
 - (void)processResponse:(BNCServerResponse *)response error:(NSError *)error {
     if (error) {
+        [BranchOpenRequest releaseOpenResponseLock];    
```

Comment:
By the time `processResponse:error:` is called, the network call has exceeded the retries.

I want to unlock the lock so that the client app won't freeze.  

But if there's a network error, Branch isn't really initialized correctly, and `getLatestReferringParamsSynchronous` will return stale information.  Maybe the `getLatestReferringParamsSynchronous` should return a dictionary like:

```
{
    error = BNCErrorNetworkError
}
```

to indicate that there was a problem?